src/inspector_agent.cc

```diff
@@ -363,20 +365,22 @@ class NodeInspectorClient : public V8InspectorClient {
     terminated_ = true;
   }
 
-  void connectFrontend(InspectorSessionDelegate* delegate) {
-    CHECK_EQ(channel_, nullptr);
-    channel_ = std::unique_ptr<ChannelImpl>(
-        new ChannelImpl(client_.get(), delegate));
+  int connectFrontend(std::unique_ptr<InspectorSessionDelegate> delegate) {
+    events_dispatched_ = true;
+    int session_id = next_session_id_++;
+    channels_[session_id] =
+        std::make_unique<ChannelImpl>(client_.get(), std::move(delegate));
```

Comment:
`client_.get()` looks kind of questionable.  Could it be turned into a `std::shared_ptr`?

Reply:
I do not want to imply shared ownership. I changed the code to use const std::unique_ptr&.